### ISSUE TITLE
covariance field plotting function as part of psyphy?

### ISSUE THREAD

**Issue by hmd101** [flatironinstitute/psyphy#56]
Should we include a convenience plotting function for covariance fields as part of the package? 
There are some speedups with jax but maplotlib is the ultimate bottleneck. 
The function `def _ellipse_segments_from_covs()` below is faster though as
all heavy math (jittering, eigvals, Cholesky, circle transform) is done
        in one batched JAX computation rather than inside a Python loop.
This could become especially relevant when we are plotting many ellipsoids, e.g., when plotting samples from the posterior.

The function should be input dim agnostic or in the very least support 2D and 3D.

```python

_THETAS = jnp.linspace(0, 2 * jnp.pi, 100)
_UNIT_CIRCLE = jnp.vstack([jnp.cos(_THETAS), jnp.sin(_THETAS)])


def _ellipse_segments_from_covs(
    centers_xy: jnp.ndarray,
    covs: jnp.ndarray,
    *,
    scale: float,
    plot_jitter: float,
    unit_circle: jnp.ndarray,
) -> tuple[jnp.ndarray, jnp.ndarray]:
    """Convert batched covariances into polyline segments for LineCollection.

    Why this is faster:
      - All heavy math (jittering, eigvals, Cholesky, circle transform) is done
        in one batched JAX computation rather than inside a Python loop.
      - Matplotlib then receives one big array (n_lines, n_pts, 2) instead of
        creating hundreds of individual Line2D artists via ax.plot.

    Returns
    -------
    segments : (n_valid, n_theta, 2)
        Polyline segments for each ellipse.
    valid_mask : (n_centers,)
        Which centers were PD after jitter.
    """
    # Add a small diagonal jitter for plotting stability.
    covs = covs + plot_jitter * jnp.eye(covs.shape[-1])

    # PD check: for a 2x2 SPD matrix, eigvalsh is cheap and reliable.
    eigvals = jnp.linalg.eigvalsh(covs)
    valid = jnp.all(eigvals > 0, axis=-1)

    # Cholesky is faster than eigendecomposition for SPD matrices.
    # We only use it for plotting ellipses (shape/orientation), not inference.
    def _cov_to_points(cov: jnp.ndarray, center: jnp.ndarray) -> jnp.ndarray:
        L = jnp.linalg.cholesky(cov)
        pts = scale * (L @ unit_circle)  # (2, n_theta)
        return (center[:, None] + pts).T  # (n_theta, 2)

    # Compute for all, then filter to valid in one shot.
    all_segments = jax.vmap(_cov_to_points)(covs, centers_xy)  # (n, n_theta, 2)
    return all_segments[valid], valid


```
